addons/website/models/ir_model.py

```diff
@@ -32,6 +33,8 @@ def get_base_url(self):
             return self.sudo().website_id.domain
         if 'company_id' in self and self.company_id.website_id.domain:
             return self.company_id.website_id.domain
+        if self.env.company.website_id.domain:
+            return self.env.company.website_id.domain
```

Comment:
I am not super familiar with our `get_base_url` usecases but I am not sure I would go in this direction. Here you are handling the context "company", but what about the context "website"? And which should come first?

I feel like `get_base_url` should be independent from the context: "give me an url to access this record, using specific record data". In case you are in a website-specific context, and you want an URL using that context, you should just call `website.get_base_url()` and not `my_record.get_base_url()`. Although... :point_down: 

Again, I may be wrong, this is not an area of the website/company system I am very familiar with. I actually **just** r+'ed a fix which is very related (I r-'ed seeing this just after :open_mouth:): https://github.com/odoo/odoo/pull/179455#issuecomment-2370989723. It's true that it would be way easier to fix that kind of bug by doing a python patch like here, and it might make sense for usecases which are in non-website apps which could need to be updated once the website is installed. @robinlej What do you think about this PR here?

Reply:
I really don't know much about multi-company environments, but indeed my 1st thought was to do something like that here:
```
        current_website = self.env['website'].get_current_website()
        if current_website:
            return current_website.domain
```

So I'm wondering if we should do that (because the current website would probably be the expected result there?). `self.env.company.website_id.domain` would still be possible as a last resort fallback if that also fails. We should test, I'm not sure how all the possibilities would work.

That said, those wouldn't solve the issue of my other PR: in the case a company is set on the product (while the product is shared cross-site) we end up on the `'company_id'` condition, and I don't think we should bypass this. So I also agree that when possible, we "should just call `website.get_base_url()` and not `my_record.get_base_url()`".

Reply:
> That said, those wouldn't solve the issue of my other PR: in the case a company is set on the product (while the product is shared cross-site) we end up on the 'company_id' condition, and I don't think we should bypass this. So I also agree that when possible, we "should just call website.get_base_url() and not my_record.get_base_url()".

True indeed, probably another good reason to not make this function here context-dependent. Seems misleading to have a function that says "in case there is a company/website use it, otherwise use the context one", when, as you explained, you probably always want to use the context one whether or not the record you are accessing has a set company/website or not... it really much depend on the usecase.

Keeping this opened for now, I need to think about it later. I'll re-r+ your other PR.